### PR TITLE
Ports of release 8.0 to the Mac, Solaris-x86, and Visual Studio 2015.

### PR DESCRIPTION
**Pull request by MichaelJCaruso**
A pull request that:

- brings release 8.0 to MacOS Sierra.
- brings release 8.0 to the Intel version of Solaris 11.3.
- insures that this version of 8.0 builds successfully on Windows 10 using Visual Studio 2015 (the community edition).

Significant changes include:

- updates to the Solaris code base to use Solaris' platform independent atomic memory operations.
- updates to the Linux and Solaris code base to prefer the use of the DCE compliant libuuid UUID generator where available.
- changes to our Visual Studio project files that allow Vision to be built directly from the directory structure found in this repository instead of first needing to rearrange that structure (project files now go to the code instead of requiring that the code first go the project files).
- many, but far from all, of the changes needed to update this code to play well with newer (and in some cases, buggier) versions of the C++ compiler and libraries.

Not included in this pull request is the integration of these changes into release 8.1; however, a git-based integration that does that has been successfully prototyped.  That prototype also gives us a simple transition to git-branch based development and release work-flows but that's a subject for future pull requests :-)

### CODE REVIEW

File: software/src/8.0/src/kernel/VkUUID.cpp [hunk 1]
```diff
@@ -58,16 +31,43 @@ typedef caddr_t ifc_buf_t;
...
+	return (uuid_t*)(&rUUID);
+    };
+    uuid_t* UCast (V::uuid_t &rUUID) {
+	return (uuid_t*)(&rUUID);
+    };
+}
+
+#elif defined(__APPLE__) || defined(__linux__) || defined(sun)
+#define USING_LIBUUID_DCE
+#include <uuid/uuid.h>
```
Comment (VCommitter) [inline software/src/8.0/src/kernel/VkUUID.cpp:58]: This isn't necessarily builtin on Linux:

```
../kernel/VkUUID.cpp:58:23: fatal error: uuid/uuid.h: No such file or directory
 #include <uuid/uuid.h>
```

is it your intention to make libuuid-devel a requirement?

**Review by VCommitter — COMMENTED** [review]
ran across the issue with libuuid-devel.  

Ran into an another problem on Solaris 10:

```
cg error (as) : "/tmp/ccfe.22374.1.s", line 24928 : redefinition of symbol "__1cBVbDVReferenceableImplementation_4n0ALThreadModelFMulti__DRTT_"
gmake: *** [Vca_VStockDeviceFactory.o] Error 1
```

`-DUSING_LEGACY_SPARC_ATOMICS` does fix this error for me.

Didn't try windows yet.

**Comment by VCommitter** [comment]
I lied.  `-DUSING_LEGACY_SPARC_ATOMICS` does not fix the redefinition error

File: software/src/8.0/src/kernel/VkUUID.cpp (hunk 1 above)
Comment (MichaelJCaruso) [reply software/src/8.0/src/kernel/VkUUID.cpp:58]: That ought to be a reasonable requirement.  libuuid is part of the Linux runtime and should provide higher quality UUID generation than what we've got.  I'm not sure why RedHat left out the part (libuuid-devel) that contains the include files.  They seem to be present elsewhere.

Don't know how much I'll be able to look, but what object files/libraries are being implicated by the linker?  Is this a clean build?

**Pull request merged**